onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -765,10 +767,18 @@ def aten_bitwise_left_shift(self: TInt, other: TInt) -> TInt:
 @torch_op("aten::bitwise_not")
 def aten_bitwise_not(self: TInt) -> TInt:
     # bitwise_not(Tensor self) -> Tensor
-
+    # TODO(titaiwang): Support BOOL input
     return op.BitwiseNot(self)
 
 
+@torch_op("aten::bitwise_not", overload=True)
+def aten_bitwise_not_bool(self: BOOL) -> BOOL:
+    # bitwise_not(Tensor self) -> Tensor
+    # FIXME(titaiwang): This is a hack to get around the fact that we don't have op.BitwiseNot supporting bool now.
```

Comment:
We will have to go with this I think. Not a hack as we need a dispatcher to handle different dtypes

Reply:
Need if for `aten_bitwise_not`?

Reply:
I think so

Reply:
Oh we can do if because the dtypes are different, _and_ we don’t know the dtype within a function

Reply:
So how should we know the dtype beforehand?

Reply:
Maybe there are only certain kinds of ops can be input of BitWise operations that we can do the processing in those ops?

Reply:
> So how should we know the dtype beforehand?

the exporter knows about dtypes so it can choose which function to use